packages/puppeteer-core/src/bidi/HTTPResponse.ts

```diff
@@ -7,86 +7,85 @@ import type * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
 import type Protocol from 'devtools-protocol';
 
 import type {Frame} from '../api/Frame.js';
-import {
-  HTTPResponse as HTTPResponse,
-  type RemoteAddress,
-} from '../api/HTTPResponse.js';
+import {HTTPResponse, type RemoteAddress} from '../api/HTTPResponse.js';
 import {UnsupportedOperation} from '../common/Errors.js';
+import {invokeAtMostOnceForArguments} from '../util/decorators.js';
 
 import type {BidiHTTPRequest} from './HTTPRequest.js';
 
 /**
  * @internal
  */
 export class BidiHTTPResponse extends HTTPResponse {
+  static create(
+    data: Bidi.Network.ResponseData,
+    request: BidiHTTPRequest
+  ): BidiHTTPResponse {
+    const response = new BidiHTTPResponse(data, request);
+    response.#initialize();
+    return response;
+  }
+
+  #data: Bidi.Network.ResponseData;
+
   #request: BidiHTTPRequest;
-  #remoteAddress: RemoteAddress;
-  #status: number;
-  #statusText: string;
-  #url: string;
-  #fromCache: boolean;
-  #headers: Record<string, string> = {};
-  #timings: Record<string, string> | null;
-
-  constructor(
-    request: BidiHTTPRequest,
-    {response}: Bidi.Network.ResponseCompletedParameters
+
+  private constructor(
+    data: Bidi.Network.ResponseData,
+    request: BidiHTTPRequest
   ) {
     super();
+    this.#data = data;
```

Comment:
Let use the same pattern you use in Request:
```suggestion
    this.#response = response;
```